### PR TITLE
DAOS-9552 pool: add performance domain as a new pool property

### PR DESCRIPTION
**Pull request by wangshilong**
Performance domain is a new pool property. Because it impacts the
algorithmic placement results and localities of all the data,
it can be only set on creation of pool and never be changed after that.
Otherwise, the DAOS stack will generate wrong layouts for objects
after the domain change

Signed-off-by: Wang Shilong <wangshilong1991@gmail.com>

### CODE REVIEW

**Comment by mjmac** [comment]
I think this PR needs to be cleanly rebased on master, as it seems to contain commits that have already landed. After this initial rebase, please refrain from using rebase again and instead merge master to get updates.

**Review by mjmac — APPROVED** [review]
Control changes look good to me.

**Comment by wangshilong** [comment]
Tests passed, ping reviewers!

**Review by mjmac — COMMENTED** [review]
Control changes look fine to me. Will leave the 2 review slots for reviewers of the C side of things.

File: src/common/proc.c [hunk 1]
```diff
@@ -94,6 +94,7 @@ crt_proc_prop_entries(crt_proc_t proc, crt_proc_op_t proc_op, daos_prop_t *prop)
 		    entry->dpe_type == DAOS_PROP_CO_OWNER ||
 		    entry->dpe_type == DAOS_PROP_PO_OWNER_GROUP ||
 		    entry->dpe_type == DAOS_PROP_CO_OWNER_GROUP ||
+		    entry->dpe_type == DAOS_PROP_PO_PERF_DOMAIN ||
```
Comment (liuxuezhao) [inline src/common/proc.c:97]: I am thinking why define PERF_DOMAIN prop type as string? 
How about take the same method as DAOS_PROP_CO_REDUN_LVL prop that defined as integer type which can directly map to "enum pool_comp_type", and the default value is PO_COMP_TP_ROOT (255), and can with similar valid value check as "C.DAOS_PROP_ENTRY_REDUN_LVL" in propHdlrMap().
By that method can make the mapping be consistent as DAOS_PROP_CO_REDUN_LVL as both them are just a domain level.
How do you think?

Comment (liuxuezhao) [reply src/common/proc.c:97]: Or allow user to input a string (valid range in "struct pool_comp_type_dict comp_type_dict::td_name"), but in C backend code probably better be same as DAOS_PROP_CO_REDUN_LVL prop that use an integer type so can directly use. 
ask user to use integer seems also looks fine and is a little simpler (just same as "--properties=rd_fac:1,rd_lvl:2" as explained at above)

Comment (wangshilong) [reply src/common/proc.c:97]: I think make it integer makes sense, thanks!

File: src/include/daos_srv/pool.h [hunk 2]
```diff
@@ -27,7 +27,7 @@
 /*
  * Aggregation of pool/container/object/keys disk format change.
  */
-#define DS_POOL_GLOBAL_VERSION		1
+#define DS_POOL_GLOBAL_VERSION		2
```
Comment (kccain) [inline src/include/daos_srv/pool.h:30]: Trying to remember all of the aspects of interop management: does DS_POOL_GLOBAL_VERSION always need to be kept at the same value as DAOS_POOL_GLOBAL_VERSION defined in include/daos/pool.h?

**Review by kccain — APPROVED** [review]

File: src/include/daos_srv/pool.h (hunk 2 above)
Comment (wangshilong) [reply src/include/daos_srv/pool.h:30]: DS_POOL_GLOBAL_VERSION is not used any more, we now use DAOS_POOL_GLOBAL_VERSION, we might need remove it for clarity.

**Review by kccain — APPROVED** [review]

**Comment by wangshilong** [comment]
ping a second review! thanks

**Review by liuxuezhao — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

**Pull request merged**